### PR TITLE
Change XmlTransform encoding to UTF-8

### PR DESCRIPTION
**Pull request by Matchile**
fixes https://github.com/nunit/nunit-console/issues/181

### CODE REVIEW

File: src/NUnitEngine/nunit.engine/Services/ResultWriters/XmlTransformResultWriter.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ public void WriteResultFile(XmlNode result, TextWriter writer)
 
         public void WriteResultFile(XmlNode result, string outputPath)
         {
-            using (var xmlWriter = new XmlTextWriter(outputPath, Encoding.Default))
+            using (var xmlWriter = new XmlTextWriter(outputPath, Encoding.UTF8))
```
Comment (mikkelbu) [inline src/NUnitEngine/nunit.engine/Services/ResultWriters/XmlTransformResultWriter.cs]: What about @jnm2's comments about using `XmlWriter.Create(new StreamWriter(outputPath))` to ensure no BOM https://github.com/nunit/nunit-console/issues/181#issuecomment-282442507. See also the note on MSDN about `XmlTextWriter` that @jnm2 linked to.

> Starting with the .NET Framework 2.0, we recommend that you create XmlWriter instances by using the XmlWriter.Create method and the XmlWriterSettings class to take advantage of new functionality.

Comment (Matchile) [reply src/NUnitEngine/nunit.engine/Services/ResultWriters/XmlTransformResultWriter.cs]: Updated to `XmlWriter.Create`. `UTF-8` should be default according to API.

File: src/NUnitEngine/nunit.engine/Services/ResultWriters/XmlTransformResultWriter.cs [hunk 2]
```diff
@@ -81,18 +81,16 @@ public void CheckWritability(string outputPath)
...
             {
-                xmlWriter.Formatting = Formatting.Indented;
                 _transform.Transform(result, xmlWriter);
             }
         }
 
         public void WriteResultFile(XmlNode result, string outputPath)
         {
-            using (var xmlWriter = new XmlTextWriter(outputPath, Encoding.Default))
+            using (var xmlWriter = XmlWriter.Create(new StreamWriter(outputPath), new XmlWriterSettings {Indent = true, ConformanceLevel = ConformanceLevel.Auto}))
```
Comment (jnm2) [inline src/NUnitEngine/nunit.engine/Services/ResultWriters/XmlTransformResultWriter.cs]: If you don't manually do `new StreamWriter`, it allows the XmlWriter to use XmlWriterSettings.CreateWriter. Does this seem like a good thing to do?

```suggestion
            using (var xmlWriter = XmlWriter.Create(outputPath, new XmlWriterSettings { Indent = true, ConformanceLevel = ConformanceLevel.Auto }))
```

**Comment by ChrisMaddock** [comment]
Let's get this merged - I'll make the last change separately. Thanks for the contribution, @Matchile! 😄

**Pull request merged**